Test/FMI2/fmi2_xml_parsing_test.c

```diff
@@ -7,29 +7,27 @@
 #include "config_test.h"
 #include "fmi_testutil.h"
 
-static const int SHOULD_NOT_LOG_EXPECTED_MSG = 0;
-static const int SHOULD_LOG_EXPECTED_MSG = 1;
 
-static int did_not_log_expected_msg;
-static char *expected_message = "Invalid structured ScalarVariable name";
-static char *name_check_test_directory;
+static int g_has_logged_expected_msg;
+static char *g_name_check_test_directory;
+static char *g_expected_message = "Invalid structured ScalarVariable name";
 
 void importlogger(jm_callbacks* c, jm_string module,
         jm_log_level_enu_t log_level, jm_string message)
 {
     printf("module = %s, log level = %d: %s\n", module, log_level, message);
-    if (!strncmp(expected_message, message, strlen(expected_message))) {
-        did_not_log_expected_msg = 0;
+    if (!strncmp(g_expected_message, message, strlen(g_expected_message))) {
+        g_has_logged_expected_msg = 1;
     }
 }
 
-void test_parser(char *xml_dir, int should_not_log_expected_msg, int configuration)
+void test_parser(char *xml_dir, int should_log_expected_msg, int configuration)
 {
     jm_callbacks cb;
     fmi_import_context_t *context;
     fmi2_import_t *fmu;
     char *full_path;
-    int res = 0;
+    int res = 1;
```

Comment:
Test was incorrectly always succeeding before.

Reply:
I would propose not assigning any value here at all and doing it more explicitly based on the result of the test.

Reply:
Yes, the test is a bit convoluted. It seems to try to handle deallocation even on test failure, making things complicated. Moving testing to catch will also make these things easier.